httpx/_models.py

```diff
@@ -843,10 +858,14 @@ def raise_for_status(self) -> None:
 
         if codes.is_client_error(self.status_code):
             message = message.format(self, error_type="Client Error")
-            raise HTTPStatusError(message, request=self.request, response=self)
+            if self._request is None:
+                raise ValueError(message)
+            raise HTTPStatusError(message, request=self._request, response=self)
```

Comment:
Raising a ValueError inside raise_for_status seems a bit off ?

Reply:
I'd suggest we start the method with something like...

```python
if self._request is None:
    raise RuntimeError("Cannot call `raise_for_status` as the request instance has not been set on this response.")
```